File: codeforces/contests/2009/E.cpp
#include <bits/stdc++.h>
using namespace std;

long long expr(long long x, long long b, long long c) {
	return x * x + x * b - c;
}

int main() {
	int _; cin >> _;
	while (_--) {
		long long n, k; cin >> n >> k;
		long long c = ((n) * (n + 2 * k - 1)) / 2, b = 2 * k - 1;

		long long x = (long long) (((double) -b + sqrt(b*b + 4 * c)) / 2.0);

		cout << min(abs(expr(x, b, c)), abs(expr(x+1, b, c))) << endl;
	}
}

File: codeforces/contests/2009/E.py
from math import sqrt

def expr(x, b, c):
	return int(x * x + x * b - c)

_ = int(input())
for __ in range(_):
	n, k = map(int, input().split())
	c = (n * (n+2*k-1)) / 2
	b = 2 * k - 1

	x = (-b + sqrt(b*b + 4 * c)) // 2
	print(min(abs(expr(x, b, c)), min(abs(expr(x+1, b, c)), abs(expr(x-1, b, c)))))